### PR TITLE
Lesson 1.4: Lifting state

### PR DESCRIPTION
**Pull request by liuba83**
This PR addresses:

- move Todo List into State
- control Add Todo Input
- add New Todo to List
- destructure props

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -4,15 +4,18 @@ import AddTodoForm from "./AddTodoForm.jsx";
 import React from "react";
 
 function App() {
-  const [newTodo, setNewTodo] = React.useState("");
   const [todoList, setTodoList] = React.useState([]);
 
+  function addTodo (newTodo) {
+    setTodoList([...todoList], newTodo)
+  }
```
Comment (liuba83) [inline src/App.jsx]: I'm not sure where the issue is, but new todo item is not added to the list. Will try to figure out

Comment (liuba83) [reply src/App.jsx]: solved

File: src/AddTodoForm.jsx [hunk 2]
```diff
@@ -1,21 +1,39 @@
-function AddTodoForm(props) {
+import React from "react";
```
Comment (in43sh) [inline src/AddTodoForm.jsx:1]: This line is not needed in modern React apps.

File: src/AddTodoForm.jsx [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
+
+function AddTodoForm({ onAddTodo }) {
+  const [todoTitle, setTodoTitle] = React.useState("");
```
Comment (in43sh) [inline src/AddTodoForm.jsx:4]: instead of importing the whole "React" you can only import useState from React.

File: src/AddTodoForm.jsx [hunk 4]
```diff
@@ -1,21 +1,39 @@
...
+
   function handleAddTodo(event) {
     event.preventDefault();
-
-    let todoTitle = event.target.elements.title.value;
+    if(todoTitle==="") {
+      alert("You have to write a todo")
+      return
+    }
     console.log(todoTitle);
```
Comment (in43sh) [inline src/AddTodoForm.jsx:17]: Using console.log() statements is fine but is not preferred.

**Review by in43sh — APPROVED** [review]
Overall LGTM! I would probably add type checking and would try to fix some little things I pointed out in the comments.

File: src/AddTodoForm.jsx (hunk 4 above)
Comment (liuba83) [reply src/AddTodoForm.jsx:17]: I'll apply this changes in next PR

**Pull request merged**